lib/ansible/modules/cloud/amazon/aws_waf_condition.py

```diff
@@ -0,0 +1,642 @@
+        missing = set(regex_pattern['regex_strings']) - set(pattern_set['RegexPatternStrings'])
+        extra = set(pattern_set['RegexPatternStrings']) - set(regex_pattern['regex_strings'])
+        if not missing and not extra:
+            return pattern_set
+        updates = [{'Action': 'INSERT', 'RegexPatternString': pattern} for pattern in missing]
+        updates.extend([{'Action': 'DELETE', 'RegexPatternString': pattern} for pattern in extra])
+        self.client.update_regex_pattern_set(RegexPatternSetId=pattern_set['RegexPatternSetId'],
+                                             Updates=updates, ChangeToken=get_change_token(self.client, self.module))
+        return self.get_regex_pattern_set_with_backoff(pattern_set['RegexPatternSetId'])['RegexPatternSet']
+
+    def delete_unused_regex_pattern(self, regex_pattern_set_id):
+        try:
+            regex_pattern_set = self.client.get_regex_pattern_set(RegexPatternSetId=regex_pattern_set_id)['RegexPatternSet']
+            updates = list()
+            for regex_pattern_string in regex_pattern_set['RegexPatternStrings']:
+                updates.append({'Action': 'DELETE', 'RegexPatternString': regex_pattern_string})
+            self.client.update_regex_pattern_set(RegexPatternSetId=regex_pattern_set_id, Updates=updates,
+                                                 ChangeToken=get_change_token(self.client, self.module))
+
+            self.client.delete_regex_pattern_set(RegexPatternSetId=regex_pattern_set_id,
+                                                 ChangeToken=get_change_token(self.client, self.module))
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self.module.fail_json_aws(e, msg='Could not delete regex pattern')
+
+    def get_condition_by_name(self, name):
+        all_conditions = [d for d in self.list_conditions() if d['Name'] == name]
+        if all_conditions:
+            return all_conditions[0][self.conditionsetid]
+
+    @AWSRetry.exponential_backoff()
+    def get_condition_by_id_with_backoff(self, condition_set_id):
+        params = dict()
+        params[self.conditionsetid] = condition_set_id
+        func = getattr(self.client, 'get_' + self.method_suffix)
+        return func(**params)[self.conditionset]
+
+    def get_condition_by_id(self, condition_set_id):
+        try:
+            return self.get_condition_by_id_with_backoff(condition_set_id)
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self.module.fail_json_aws(e, msg='Could not get condition')
+
+    def list_conditions(self):
+        method = 'list_' + self.method_suffix + 's'
+        try:
+            paginator = self.client.get_paginator(method)
+            func = paginator.paginate().build_full_result
+        except botocore.exceptions.OperationNotPageableError:
```

Comment:
If an older version of botocore is installed a KeyError is raised instead here and it took me a while to figure out why. Might want to document the minimum version.

Reply:
I just hit this myself! I'll handle the KeyError too.